ote/src/cljs/ote/views/admin/interfaces.cljs

```diff
@@ -13,96 +13,97 @@
-       [:div.col-md-6 [:a {:href     "#"
-                           :on-click #(do
-                                        (.preventDefault %)
-                                        (e! (fp/->ChangePage :transport-operator {:id (:operator-id interface)})))}
-                       (:operator-name interface)]]]
-      [:div.row
-       [:div.col-md-4 (stylefy/use-style style-admin/modal-data-label) "Puhelin: "]
-       [:div.col-md-6 (:operator-phone interface)]]
-      [:div.row
-       [:div.col-md-4 (stylefy/use-style style-admin/modal-data-label) "GSM: "]
-       [:div.col-md-6 (:operator-gsm interface)]]
-      [:div.row
-       [:div.col-md-4 (stylefy/use-style style-admin/modal-data-label) "Sähköposti: "]
-       [:div.col-md-6 (:operator-email interface)]]]
-
-     [:div.col-md-6 {:style {:padding-left "20px"}}
-      [:div.row [:h2 "Rajapinta kuuluu palveluun:"]]
-      [:div.row
-       [:div.col-md-4 (stylefy/use-style style-admin/modal-data-label) "Nimi: "]
-       [:div.col-md-6 [:a {:href     "#"
-                           :on-click #(do
-                                        (.preventDefault %)
-                                        (e! (fp/->ChangePage :edit-service {:id (:service-id interface)})))}
-                       (:service-name interface)]]]
-      [:div.row
-       [:div.col-md-4 (stylefy/use-style style-admin/modal-data-label) "Puhelin: "]
-       [:div.col-md-6 (:service-phone interface)]]
-      [:div.row
-       [:div.col-md-4 (stylefy/use-style style-admin/modal-data-label) "Sähköposti: "]
-       [:div.col-md-6 (:service-email interface)]]]]))
+    (when interface
+      [ui/dialog
+       {:open true
+        :actionsContainerStyle style-dialog/dialog-action-container
+        :modal false
+        :auto-scroll-body-content true
+        :on-request-close #(e! (admin-controller/->CloseInterfaceErrorModal))
+        :title title
+        :actions [(r/as-element
+                    [ui/flat-button
+                     {:label (tr [:buttons :close])
+                      :secondary true
+                      :primary true
+                      :on-click #(e! (admin-controller/->CloseInterfaceErrorModal))}])]}
+       [:div.col-md-8
+        [:div.row
+         [:div.col-md-6 (stylefy/use-style style-admin/modal-data-label) "Rajapinnan osoite: "]
+         [:div.col-md-6 [linkify (:url interface) (:url interface) {:target "_blank"}]]]
+        [:div.row
+         [:div.col-md-6 (stylefy/use-style style-admin/modal-data-label) "Rajapinnan tyyppi: "]
+         [:div.col-md-6 (str/join ", " (:format interface))]]
+        [:div.row
+         [:div.col-md-6 (stylefy/use-style style-admin/modal-data-label) "Virhe: "]
+         [:div.col-md-6
+          (when (:import-error interface)
+            [:p (:import-error interface)])
+          (when (:db-error interface)
+            [:p (:db-error interface)])]]]])))
 
+(defn operator-modal [e! app]
```

Comment:
Modals are an old component so maybe we shouldn't spend too much time on them, but still I was thinking if they should be replaced with some other approach. Maybe with flex, but nevermind for now :)